### PR TITLE
WFCORE-1841 lazy subsystem parser support

### PR DESCRIPTION
**Pull request by ctomc**
- introduces parser providers (via staxmapper 1.3)
- migrates core subsystems to use it
- should reduce memory usage down a bit

### CODE REVIEW

File: controller/src/main/java/org/jboss/as/controller/PersistentResourceXMLDescription.java [hunk 1]
```diff
@@ -462,7 +463,8 @@ public static PersistentResourceXMLBuilder builder(final PathElement pathElement
         protected boolean noAddOperation;
         protected AdditionalOperationsGenerator additionalOperationsGenerator;
         protected final LinkedList<AttributeDefinition> attributeList = new LinkedList<>();
-        protected final List<PersistentResourceXMLBuilder> children = new ArrayList<>();
+        final List<PersistentResourceXMLBuilder> childrenBuilder = new ArrayList<>();
+        final List<PersistentResourceXMLDescription> children = new ArrayList<>();
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/PersistentResourceXMLDescription.java]: This changes the visibility of 'children'.

It's too bad this class isn't final and all of these aren't just private. There are no subclasses in core or full.

File: controller/src/main/java/org/jboss/as/controller/PersistentResourceXMLDescription.java [hunk 2]
```diff
@@ -462,7 +463,8 @@ public static PersistentResourceXMLBuilder builder(final PathElement pathElement
         protected boolean noAddOperation;
         protected AdditionalOperationsGenerator additionalOperationsGenerator;
         protected final LinkedList<AttributeDefinition> attributeList = new LinkedList<>();
-        protected final List<PersistentResourceXMLBuilder> children = new ArrayList<>();
+        final List<PersistentResourceXMLBuilder> childrenBuilder = new ArrayList<>();
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/PersistentResourceXMLDescription.java]: Nit: 'childrenBuilders' is a better name as this is a collection.

File: controller/src/main/java/org/jboss/as/controller/parsing/ExtensionParsingContext.java [hunk 3]
```diff
@@ -66,6 +67,9 @@
      */
     void setSubsystemXmlMapping(String subsystemName, String namespaceUri, XMLElementReader<List<ModelNode>> reader);
 
+
+    void setSubsystemXmlMapping(String subsystemName, String namespaceUri, Supplier<XMLElementReader<List<ModelNode>>> supplier);
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/parsing/ExtensionParsingContext.java]: Needs verbose javadoc.

Comment (ctomc) [inline controller/src/main/java/org/jboss/as/controller/parsing/ExtensionParsingContext.java]: It was written, but lost in refactoring somewhere. 
fixed.

File: controller/src/main/java/org/jboss/as/controller/PersistentResourceXMLDescription.java (hunk 2 above)
Comment (ctomc) [inline controller/src/main/java/org/jboss/as/controller/PersistentResourceXMLDescription.java]: fixed

**Comment by bstansberry** [comment]
retest this please

**Pull request merged**